addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/internal/state/MonitorThingState.java

```diff
@@ -0,0 +1,537 @@
+                    break;
+
+                case PRE_ALARM:
+                case ALARM:
+                case ALERT:
+                    alarmedDetailedState.set(true);
+                    recordingDetailedState.set(true);
+                    break;
+
+                case RECORDING:
+                    alarmedDetailedState.set(false);
+                    recordingDetailedState.set(true);
+                    break;
+                case UNKNOWN:
+                default:
+                    alarmedDetailedState.set(false);
+                    recordingDetailedState.set(false);
+                    break;
+
+            }
+            logger.debug(
+                    "{}: Recalculate channel states based on Detailed State: DetailedState='{}' -> alarmState='{}', recordingState='{}'",
+                    logIdentifier, monitorStatus.name(), alarmedDetailedState, recordingDetailedState);
+
+            if (monitorStatus == ZoneMinderMonitorStatusEnum.IDLE && !alarmedDetailedState.get()
+                    && activeEvent != null) {
+                activeEvent = null;
+            }
+
+            updateEventChannels();
+
+            // Now we can conclude on the Alarmed and Recording channel state
+            setMonitorRecording((recordingFunction.get() && recordingDetailedState.get() && isEnabled()));
+            setMonitorEventState((alarmedFunction.get() && alarmedDetailedState.get() && isEnabled()));
+
+            switch (getMonitorDetailedStatus()) {
+                case UNKNOWN:
+                    setMonitorEventCause(UnDefType.UNDEF);
+                    setMonitorEventMotion(UnDefType.UNDEF);
+                    break;
+
+                case PRE_ALARM:
+                case ALARM:
+                case ALERT:
+                    if (activeEvent != null) {
+                        setMonitorEventCause(new StringType(activeEvent.getCause()));
+                        setMonitorEventMotion(
+                                activeEvent.getCause().equalsIgnoreCase("motion") ? OnOffType.ON : OnOffType.OFF);
+                    } else {
+                        setMonitorEventMotion(OnOffType.OFF);
+                    }
+                    break;
+
+                case IDLE:
+                case RECORDING:
+                default:
+                    setMonitorEventCause(new StringType(""));
+                    setMonitorEventMotion(OnOffType.OFF);
+                    break;
+            }
+        } catch (Exception ex) {
+            logger.error("{}: context='recalculate' Exception occurred", logIdentifier, ex);
+        } finally {
+            isDirty = false;
+        }
+    }
+
+    protected State getChannelByteArrayAsRawType(ByteArrayOutputStream image) {
+        State state = UnDefType.UNDEF;
+        try {
+            if (image != null) {
+                state = new RawType(image.toByteArray(), "image/jpeg");
+            }
+
+        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.